src/pages/graphile-build/all-hooks.md

```diff
@@ -10,54 +10,68 @@ The following hooks are currently supported, but more may be added in future.
 Trying to add a hook for a hook name that does not exist will result in an
 error.
 
-[(See hooks in the source)](https://github.com/graphile/graphile-engine/blob/996e28f0af68f53e264170bd4528b6500ff3ef25/packages/graphile-build/SchemaBuilder.js#L11-L59)
+[(See hooks in the source)](https://github.com/graphile/graphile-engine/blob/v4.4.4/packages/graphile-build/src/SchemaBuilder.js#L167-L230)
+
+- `inflection`: The inflection object provides methods to derive names and will be part of the Build object,
+  hook on the 'inflection' event to customize this object.
 
 - `build`: The build object represents the current schema build and is passed
   to all hooks, hook the 'build' event to extend this object.
 
 - `init`: The init event is triggered after `build` (which should not generate
   any GraphQL objects) and can be used to build common object types that may be
-  useful later. The argument to this is an empty object and should be passed
-  through unmodified (it is ignored currently).
-
-- `GraphQLSchema`: This event defines the root-level schema; hook it to add `query`,
-  `mutation`, `subscription` or similar GraphQL fields.
+  useful later. The argument to this is an opaque object and should be passed
+  through unmodified (it currently is an empty object that gets ignored).
 
 - `GraphQLObjectType*`: When creating a GraphQLObjectType via
-  `newWithHooks`, we'll execute, the following hooks:
-
-  - `GraphQLObjectType` to add any root-level attributes, e.g. add a description
-  - `GraphQLObjectType:interfaces` to add additional interfaces to this object type
-  - `GraphQLObjectType:fields` to add additional fields to this object type (is
-    ran asynchronously and gets a reference to the final GraphQL Object as
-    `Self` in the context)
-  - `GraphQLObjectType:fields:field`: to add any root-level attributes to an
+  `newWithHooks`, we'll execute the following hooks:
+
+  - `GraphQLObjectType` to add/remove any root-level attributes, e.g. a description
+  - `GraphQLObjectType:interfaces` to add additional interfaces to this object type or remove them
+  - `GraphQLObjectType:fields` (*deferred*) to add additional fields to this object type or remove them.
```

Comment:
I just kept it - removing is currently mentioned at https://www.graphile.org/graphile-build/hooks/#which-hook-to-attach-to-hookname. Also I think the graphile-engine docs should mention this separately from postgraphile/extending-raw - but it might be a good idea to add the warning about adding things back, including an explanation of how lookahead is affected (I'm not sure I understand this).

Reply:
Many people will only scan this section; I'd rather that it said about "adding fields" and didn't say about removing them - that way they can wonder about why removing isn't there and find that elsewhere with the full explanation. Removing fields is not an intended behaviour; it works but I would strongly advise against it.

Reply:
OK